Automattic/vip-go-mu-plugins
# Search: Add index settings monitoring

## Description

Adds index setting monitoring to our existing health job - this will detect and log discrepancies between the desired and current index settings. In the future, we can automatically fix settings.

Depends on https://github.com/Automattic/ElasticPress/pull/81

## Changelog Description

### Update: Add monitoring of Search index settings

This change adds a new task to Search's health cron job that detects and reports any discrepancies between the desired and actual settings for each index.

## Checklist

Please make sure the items below have been covered before requesting a review:

- n/a This change has relevant documentation additions / updates (if applicable).

## Steps to Test

Rather than wait for this to actually run via cron, and since it's feature-gated, we can run the cron job manually:

1. Add a `var_dump()` to `HealthJob::check_all_indexables_settings_health()` - `var_dump( $unhealthy_indexables );` - this is a super hack but this method sends alerts to Slack / IRC which doesn't work in local dev or sandbox (need a better way to do this)
1.`wp eval "(new \Automattic\VIP\Search\HealthJob(\Automattic\VIP\Search\Search::instance()))->check_all_indexables_settings_health();"`
1. Depending on your settings, this may or may not produce any output
1. If it doesn't, you can change a setting. The easiest way is to change the value returned from `Search::filter__ep_default_index_number_of_replicas()` from `2` to anything else. Alternatively you can implement `ep_default_index_number_of_replicas` to return some other number in your theme
1. Once there is a discrepancy between what VIP Search thinks the settings should be and what they are, you'll see it in the `var_dump()` output
